src/less/components/search/index.less

```diff
@@ -1,4 +1,117 @@
 @import (reference) 'mixins';
+@import (reference) './SearchControl';
+
+// ----- LAYOUT -----
+/*
+    auto    1fr    auto
+  +-------+-------+-------+
+  | lbl   | lbl   | lbl   | auto
+  +-------+-------+-------+
+  | icon  | input | btn   | auto
+  +-------+-------+-------+
+  | extra | extra | extra | auto
+  +-------+-------+-------+
+*/
+
+
+hx-search-control {
+  -ms-grid-column-align: start;
+  -ms-grid-columns: 2rem 1fr auto;
+  -ms-grid-rows: auto auto auto;
+  display: -ms-grid;
+  display: grid;
+  grid-template-areas:
+    'lbl   lbl   lbl'
+    '.     input btn'
+    'extra extra extra';
+  grid-template-columns: 2rem 1fr auto;
+  grid-template-rows: auto auto auto;
+
+  > input[type="search"] {
+    -moz-appearance: textfield;
+    -webkit-appearance: textfield;
+    -ms-grid-column: 2;
+    -ms-grid-row: 2;
+    background: none;
+    border: none;
+    font-size: 1rem;
+    grid-area: input;
+    height: 2rem;
+    min-width: 8rem; // ~16 chars
+    z-index: 1;
+
+    // reset for IE as well
+    &::-ms-clear {
+      display: none;
+      height: 0;
+      width : 0;
+    }
+
+
+    &::-webkit-search-cancel-button {
+      display: none;
+    }
+
+    &:focus {
+      outline: 0;
+    }
+  }
+
+  > input[type="search"]:focus ~ hx-search,
+  > button.hxClear:focus ~ hx-search {
+    border-color: @cyan-700;
+    box-shadow: @focus-glow;
+  }
+
+  button {
+    -ms-grid-column: 3;
+    -ms-grid-row: 2;
+    background-color: transparent;
+    border: none;
+    color: @gray-700;
+    cursor: pointer;
+    font-size: 1rem;
+    grid-area: btn;
+    line-height: 1;
+    outline: 0;
+    padding: 0.5rem;
+    z-index: 1;
+
+    &:focus {
+      outline: 1px dotted @gray-700;
+      outline-offset: -6px;
+    }
+
+    &::-moz-focus-inner {
+      border: 0;
+      outline: 0;
+    }
+  }
+
+  hx-search {
+    -ms-grid-column-span: 3;
+    -ms-grid-row: 2;
+    background-color: @gray-0;
+    border-radius: 2px;
+    border: 1px solid @gray-500;
+    grid-column: ~"1 / span 3";
```

Comment:
Do we still need the `~" ... "` workaround? Originally it was used to prevent LESS from calculating `1 / 3`, but now that there's a `span` in the mix, it shouldn't try to calculate.

Reply:
I just checked. The `~"..."` workaround isn't necessary given that we have `span` in the mix. LESS won't try to calculate `1 / 3`. Go ahead and remote the `~"  "` surrounding the CSS value;

Reply:
Ok, I removed the workaround.